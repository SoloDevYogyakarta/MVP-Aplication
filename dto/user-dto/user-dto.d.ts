declare class LoginField {
    token: string;
    password: string;
    secret: string;
}
declare class RegisterField {
    plat_number: string;
    phone_number: string;
    name: string;
    motor: string;
    year_production: number;
    address: string;
    password: string;
    confirmation: string;
}
declare class ChangePasswordField {
    plat_number: string;
    old_password: string;
    password: string;
    confirmation: string;
}
export { LoginField, RegisterField, ChangePasswordField };
