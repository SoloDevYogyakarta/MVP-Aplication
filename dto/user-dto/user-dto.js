"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangePasswordField = exports.RegisterField = exports.LoginField = void 0;
const class_validator_1 = require("class-validator");
class LoginField {
    constructor() {
        this.secret = 'SECRET';
    }
}
exports.LoginField = LoginField;
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], LoginField.prototype, "token", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], LoginField.prototype, "password", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    (0, class_validator_1.IsOptional)(),
    __metadata("design:type", String)
], LoginField.prototype, "secret", void 0);
class RegisterField {
}
exports.RegisterField = RegisterField;
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], RegisterField.prototype, "plat_number", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], RegisterField.prototype, "phone_number", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], RegisterField.prototype, "name", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    (0, class_validator_1.IsOptional)(),
    __metadata("design:type", String)
], RegisterField.prototype, "motor", void 0);
__decorate([
    (0, class_validator_1.IsNumber)(),
    __metadata("design:type", Number)
], RegisterField.prototype, "year_production", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], RegisterField.prototype, "address", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    (0, class_validator_1.IsOptional)(),
    __metadata("design:type", String)
], RegisterField.prototype, "password", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    (0, class_validator_1.IsOptional)(),
    __metadata("design:type", String)
], RegisterField.prototype, "confirmation", void 0);
class ChangePasswordField {
}
exports.ChangePasswordField = ChangePasswordField;
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], ChangePasswordField.prototype, "plat_number", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], ChangePasswordField.prototype, "old_password", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], ChangePasswordField.prototype, "password", void 0);
__decorate([
    (0, class_validator_1.IsString)(),
    __metadata("design:type", String)
], ChangePasswordField.prototype, "confirmation", void 0);
//# sourceMappingURL=user-dto.js.map