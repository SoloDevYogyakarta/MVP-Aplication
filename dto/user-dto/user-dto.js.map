{"version":3,"file":"user-dto.js","sourceRoot":"","sources":["../../../src/dto/user-dto/user-dto.ts"],"names":[],"mappings":";;;;;;;;;;;;AAAA,qDAAiE;AAEjE,MAAM,UAAU;IAAhB;QASE,WAAM,GAAW,QAAQ,CAAC;IAC5B,CAAC;CAAA;AA6CQ,gCAAU;AArDjB;IADC,IAAA,0BAAQ,GAAE;;yCACG;AAGd;IADC,IAAA,0BAAQ,GAAE;;4CACM;AAIjB;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;0CACa;AAG5B,MAAM,aAAa;CA2BlB;AAgBoB,sCAAa;AAzChC;IADC,IAAA,0BAAQ,GAAE;;kDACS;AAGpB;IADC,IAAA,0BAAQ,GAAE;;mDACU;AAGrB;IADC,IAAA,0BAAQ,GAAE;;2CACE;AAIb;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;4CACC;AAGd;IADC,IAAA,0BAAQ,GAAE;;sDACa;AAGxB;IADC,IAAA,0BAAQ,GAAE;;8CACK;AAIhB;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;+CACI;AAIjB;IAFC,IAAA,0BAAQ,GAAE;IACV,IAAA,4BAAU,GAAE;;mDACQ;AAGvB,MAAM,mBAAmB;CAYxB;AAEmC,kDAAmB;AAZrD;IADC,IAAA,0BAAQ,GAAE;;wDACS;AAGpB;IADC,IAAA,0BAAQ,GAAE;;yDACU;AAGrB;IADC,IAAA,0BAAQ,GAAE;;qDACM;AAGjB;IADC,IAAA,0BAAQ,GAAE;;yDACU"}