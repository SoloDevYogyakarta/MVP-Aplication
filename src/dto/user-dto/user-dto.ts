import { IsNumber, IsOptional, IsString } from 'class-validator';

class LoginField {
  @IsString()
  token: string;

  @IsString()
  password: string;

  @IsString()
  @IsOptional()
  secret: string = 'SECRET';
}

class RegisterField {
  @IsString()
  plat_number: string;

  @IsString()
  phone_number: string;

  @IsString()
  name: string;

  @IsString()
  @IsOptional()
  motor: string;

  @IsNumber()
  year_production: number;

  @IsString()
  address: string;

  @IsString()
  @IsOptional()
  password: string;

  @IsString()
  @IsOptional()
  confirmation: string;
}

class ChangePasswordField {
  @IsString()
  plat_number: string;

  @IsString()
  old_password: string;

  @IsString()
  password: string;

  @IsString()
  confirmation: string;
}

export { LoginField, RegisterField, ChangePasswordField };
