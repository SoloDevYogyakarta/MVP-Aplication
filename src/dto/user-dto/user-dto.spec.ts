import { faker } from '@faker-js/faker';
import { validate } from 'class-validator';
import * as UserDto from './user-dto';

describe('UserDto', () => {
  it('should to be defined', () => expect(UserDto).toBeDefined());

  it('render correctly', () => expect(UserDto).toMatchSnapshot());

  it('LoginField', async () => {
    const data = new UserDto.LoginField();
    data.token = `B ${Math.floor(1000 + Math.random() * 9999)} HSA`;
    data.password = 'password';
    const valid = await validate(data);
    expect(valid.length).toEqual(0);
  });

  it('RegisterField', async () => {
    const data = new UserDto.RegisterField();
    data.plat_number = `B ${Math.floor(1000 + Math.random() * 9999)} HSA`;
    data.phone_number = faker.phone.number();
    data.name = faker.person.fullName();
    data.motor = faker.commerce.productName();
    data.year_production = Math.floor(1000 + Math.random() * 2023);
    data.address = faker.location.streetAddress();
    data.password = 'password';
    data.confirmation = 'password';
    const valid = await validate(data);
    expect(valid.length).toEqual(0);
  });

  it('ChangePasswordField', async () => {
    const data = new UserDto.ChangePasswordField();
    data.plat_number = `B ${Math.floor(1000 + Math.random() * 9999)} HSA`;
    data.old_password = 'password';
    data.password = 'password';
    data.confirmation = 'password';
    const valid = await validate(data);
    expect(valid.length).toEqual(0);
  });
});
