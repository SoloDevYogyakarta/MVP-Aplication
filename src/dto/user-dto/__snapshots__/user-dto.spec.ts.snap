// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDto render correctly 1`] = `
{
  "ChangePasswordField": [Function],
  "LoginField": [Function],
  "RegisterField": [Function],
}
`;
